tools/realsense-viewer/auto_updater/http_downloader.cpp

```diff
@@ -0,0 +1,116 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <sstream>
+#include <iostream>
+#include <thread>
+#include <atomic>
+#include <mutex>
+
+#include <curl/curl.h>
+#include <curl/easy.h>
+
+#include "http_downloader.h"
+
+namespace rs2
+{
+    std::mutex initialize_mutex;
```

Comment:
Global mutex is risky. 
Use layers singleton if global sync is necessary:
http://laristra.github.io/flecsi/src/developer-guide/patterns/meyers_singleton.html

Reply:
http_downloader is built in a way it can be used with multiple instances, each one can download it's required file with it's own callback function, 
The only thing that needs to be protected (according to libcurl documentation ) is "curl_easy_init" function, as it sets global system settings.
libcurl also document not to share curl handler object across thread as it is not locking the handler functions.
With the global mutex I protected the global settings and allow the user to create multiple instances of downloaders each one with it's own logic.
Is this answers about the locking issue?